tests/regression/13-privatized/89-write-lacking-precision.c

```diff
@@ -0,0 +1,30 @@
+// PARAM: --set ana.base.privatization write
+#include<pthread.h>
+struct a {
+  char* b;
+};
+
+struct a *c;
+struct a h = {""};
+struct a i = {"string"};
+
+void* d(void* args) {
+  struct a r;
+  if (c->b) {
+    __goblint_check(strlen(h.b) == 0); // Should also work for write!
```

Comment:
I'm confused about what's happening here. It's checking if `c->b` is a non-NULL byte, so the true branch is taken if `c == &i`. But then the check is on `h` directly, not `c->b`.
The `if` seems completely irrelevant to the check which would also pass without it.